.github/workflows/deploy.yaml

```diff
@@ -25,10 +25,25 @@ jobs:
       - name: Run tests
         run: |
           poetry run pytest -r a src tests --doctest-modules
-      - name: Publish to PyPI
-        env:
-          PYPI_TOKEN: "${{ secrets.PYPI_TOKEN }}"
 
-        run: |
-          poetry config pypi-token.pypi $PYPI_TOKEN
-          poetry publish --build --no-interaction
+  upload-all:
+    name: Upload if release
+    needs: [ build_wheels, build_sdist, pre-deploy-checks ]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
I don't think we need this, because the whole workflow file has this restriction (or do I miss something?)